### ISSUE TITLE
[BUG]: Group column labels not recognized in Classifier

### ISSUE THREAD

**Issue by drewaight** [pycaret/pycaret#3071]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

This issue was fixed for the regressor in [#3057](https://github.com/pycaret/pycaret/pull/3057). For the classifier neither external cvs nor fold_groups for groupkfold is working as intended

![image](https://user-images.githubusercontent.com/53615928/200351885-98cf2568-fbae-44cc-b58f-f220bd6d6897.png)
![image](https://user-images.githubusercontent.com/53615928/200351981-c0fdc46e-1a11-4885-a49b-6375e038f635.png)

![image](https://user-images.githubusercontent.com/53615928/200352238-ae548eb2-44a7-43f8-ba20-29161aca28c4.png)
![image](https://user-images.githubusercontent.com/53615928/200352288-5bccb772-2445-4b35-8e8c-28d8da161875.png)

### Reproducible Example

```python
from pycaret.classification import *

model = setup(data=X, target=targ, fold_strategy='groupkfold', fold_groups='labels', ignore_features='labels')

compare_models(n_select = 5, verbose=False)
```

### Expected Behavior

groupkfold recognizes column labels.
external scikitlearn splitter is recognized.

### Actual Results

```python-traceback
ValueError: Invalid value for the fold_groups parameter. Column labels is not present in the dataset.
```

### Installed Versions

2.3.10
3.0.0rc3
3.0.0rc4

**Comment by Ali-Flt** [comment]
There has been a recent PR on groupkfold #3057.
Did it not work even by installing from the master branch?
`pip install -U git+https://github.com/pycaret/pycaret.git@master` 
It is possible that groupkfold still has bugs so for now try `fold_groups=df_train['labels']`

**Comment by drewaight** [comment]
Confirmed works with regression and 3.0.0.rc4 (fresh installed as suggested above)

![image](https://user-images.githubusercontent.com/53615928/200427547-f53a54d9-fa95-4917-8992-726ee0003474.png)

Classification gives "Column labels is not present in the dataset". Referencing dataframe X['labels'] (or variants thereof) gives the "empty compare models result" 

![image](https://user-images.githubusercontent.com/53615928/200427882-ace13d2e-c00c-42c3-a2a0-3344f63f831a.png)

It seems like the bug was fixed for regression but just not applied to the classification module.

**Comment by Ali-Flt** [comment]
Even for the regression module after running `compare_models()`, `automl()` fails for me (when giving a string as `fold_groups`). It appears that the groupkfold strategy code is very buggy at the moment.

**Comment by moezali1** [comment]
@tvdboom Seems like related to the issue you were working on. Mind taking a look?

**Comment by tvdboom** [comment]
I fixed the bug in #3070. Thanks for pointing it out. I tested it for both classification and regression, fold_groups as str and as array. `automl` also works now
